packages/store/src/patterns/interface-tools.js

```diff
@@ -251,3 +327,182 @@ const emptyRecord = harden({});
  * @returns {{}}
  */
 export const initEmpty = () => emptyRecord;
+
+/**
+ * @template [S = any]
+ * @template [T = any]
+ * @typedef {object} Context
+ * @property {S} state
+ * @property {T} self
+ */
+
+/**
+ * @template A,S,T
+ * @param {string} tag
+ * @param {any} interfaceGuard
+ * @param {(...args: A[]) => S} init
+ * @param {T} methods
+ * @param {object} [options]
+ * @returns {(...args: A[]) => (T & RemotableBrand<{}, T>)}
+ */
+/**
+ * It takes a bunch of
+ * arguments and returns a function that creates an object with a prototype
+ * that has methods that can access the object's state
+ * @param tag - a string that will be used to identify the class in error messages
+ * @param interfaceGuard - a predicate that takes an object and returns true if it is an instance of
+ * the class.
+ * @param init - a function that takes the arguments passed to the constructor and returns the initial
+ * state record.
+ * @param methods - a record of methods that will be added to the prototype of the class.
+ * @param [options] - an optional object with a finish method that is called after the instance is
+ * created.
+ * @returns A function that takes a list of arguments and returns an object.
+ */
```

Comment:
Does it work to declare the params and returns twice?   I'd expect it to work better if the comments from the second iteration were just merged into the first iteration, which has the types.
```suggestion
/**
 * It takes a bunch of
 * arguments and returns a function that creates an object with a prototype
 * that has methods that can access the object's state
 * @template A,S,T
 * @param {string} tag - a string that will be used to identify the class in error messages
 * @param {any} interfaceGuard - a predicate that takes an object and returns true if it
 *    is an instance of the class.
 * @param {(...args: A[]) => S} init - a function that takes the arguments passed to the
 *     constructor and returns the initial state record.
 * @param {T} methods - a record of methods that will be added to the prototype of the class.
 * @param {object} [options] - an optional object with a finish method that is called after
 *    the instance is created.
 * @returns {(...args: A[]) => (T & RemotableBrand<{}, T>)} A function that takes a list of
 *   arguments and returns an object.
 */
```

Reply:
sure looks better with just the one comment - I'm just reluctant to delete any of the team's hard work! =)

Reply:
Hi @Chris-Hibbert and @cboydstun , I moved @cboydstun 's changes to https://github.com/Agoric/agoric-sdk/pull/6123 , which I'll treat as a source of suggestions of changes to bring back here. In the meantime, for comments here on those changes, it isn't clear what still applies to this PR without them, so I'll leave those conversations resolved. If you see a comment that does apply to the new state of this PR, please add it back as a fresh comment.